Assets/Scripts/Gameplay/Player.cs

```diff
@@ -54,29 +61,53 @@ void Update()
             HandleOrientation();
 
             _animator.SetBool(IsJumpingID, _isJumping);
-            if (PlayerInput.Jump && !_isJumping)
+            if (isPlayer && PlayerInput.Jump && !_isJumping)
             {
-                Jump();
+                if (energy > 0)
+                {
+                    Jump();
+                }
             }
 
-            if (PlayerInput.Fire && CanAttack())
+            if (isPlayer && PlayerInput.Fire && CanAttack())
             {
                 _animator.SetTrigger(AttackID);
                 _weapon.Fire();
             }
+
+            if (isPlayer && PlayerInput.ResetEnergy)
+            {
+                energy = 100f;
+                UpdateEnergyLabel();
+            }
             _weapon.IncrementTimer(Time.deltaTime);
         }
 
         void MoveHorizontal()
         {
-            _body.velocity = new Vector2(PlayerInput.Horizontal * speed, _body.velocity.y);
-            _animator.SetBool(IsRunningID, PlayerInput.Horizontal != 0);
+            if (CanMove() && isPlayer)
```

Comment:
This `isPlayer` check and the one for jumping is breaking the unit tests. When you added these was something actively breaking or were you trying to make it more correct? I think we need to add a player controller object, I just added `isPlayer` because my move input was moving all the sprites when I was testing combat

Reply:
There were errors every time there was a `PlayerInput` check because it's only instantiated when `isPlayer` is true in the awake method. So this ignores it all for the non-player character, but we'd want to fix this all up to support multiple players/AI knowing where to go.